## Final Project Submission

Please fill out:
* Student name: Berke Tezcan
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: James Irving
* Blog post URL:


## Importing Data


In [109]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import os

In [110]:
folder ='zippedData/'
os.listdir(folder)

['bom.movie_gross.csv.gz',
 'imdb.name.basics.csv.gz',
 'imdb.title.akas.csv.gz',
 'imdb.title.basics.csv.gz',
 'imdb.title.crew.csv.gz',
 'imdb.title.principals.csv.gz',
 'imdb.title.ratings.csv.gz',
 'rt.movie_info.tsv.gz',
 'rt.reviews.tsv.gz',
 'tmdb.movies.csv.gz',
 'tn.movie_budgets.csv.gz']

In [111]:
bom_movie_gross = pd.read_csv(f'{folder}bom.movie_gross.csv.gz')
imdb_name_basics = pd.read_csv(f'{folder}imdb.name.basics.csv.gz')
imdb_title_akas = pd.read_csv(f'{folder}imdb.title.akas.csv.gz')
imdb_title_basics = pd.read_csv(f'{folder}imdb.title.basics.csv.gz')
imdb_title_crew = pd.read_csv(f'{folder}imdb.title.crew.csv.gz')
imdb_title_principals = pd.read_csv(f'{folder}imdb.title.principals.csv.gz')
imdb_title_ratings = pd.read_csv(f'{folder}imdb.title.ratings.csv.gz')
rt_movie_info = pd.read_csv(f'{folder}rt.movie_info.tsv.gz', delimiter='\t')
rt_reviews = pd.read_csv(f'{folder}rt.reviews.tsv.gz', delimiter='\t', encoding='latin1')
tmdb_movies = pd.read_csv(f'{folder}tmdb.movies.csv.gz',index_col=0)
tn_movie_budgets = pd.read_csv(f'{folder}tn.movie_budgets.csv.gz')

In [112]:
tmdb_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [113]:
tmdb_movies['id'].value_counts()

292086    3
463839    3
11976     3
391872    3
416572    3
         ..
356987    1
350846    1
479871    1
500353    1
524288    1
Name: id, Length: 25497, dtype: int64

In [114]:
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [115]:
bom_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [116]:
imdb_name_basics.head()

nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

In [117]:
imdb_title_akas.head()

title_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   

  language        types   attributes  is_original_title  
0       bg          NaN          NaN                0.0  
1      NaN  imdbDisplay          NaN                0.0  
2      NaN  imdbDisplay          NaN                0.0  
3      NaN          NaN  short title                0.0  
4      NaN  imdbDisplay          NaN                0.0

In [118]:
imdb_title_principals.head()

tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

In [119]:
imdb_title_crew.head()

tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

In [120]:
imdb_title_ratings.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [121]:
rt_movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [122]:
rt_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [123]:
tmdb_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [124]:
imdb_title_basics.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

# Release Date vs. Profit

### EDA to determine which movie df to use (looking at Sample Size, n)

In [125]:
#EDA to figure out which dataframe has most movies for the past decade 2010 on excluding 2020.

movies_financials = pd.merge(imdb_title_basics, bom_movie_gross, how="left", left_on='primary_title', right_on='title')
movies_financials.isna().sum()
# len(tmdb_movies) #26517
# len(bom_movie_gross) #3387
# len(imdb_title_basics) #146144


tconst                  0
primary_title           0
original_title         21
start_year              0
runtime_minutes     31739
genres               5408
title              142780
studio             142783
domestic_gross     142804
foreign_gross      144103
year               142780
dtype: int64

In [126]:
imdb_title_basics[(imdb_title_basics['start_year']>=2010) & (imdb_title_basics['start_year']!=2020)] #937 rows in 2020


tconst                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                   NaN  
146143              NaN           Documentary  

[145207 rows x 6 columns]

In [127]:
imdb_title_basics['start_year'].min()

2010

In [128]:
imdb_title_basics['start_year'].unique()

array([2013, 2019, 2018, 2017, 2012, 2010, 2011, 2015, 2021, 2016, 2014,
       2020, 2022, 2023, 2024, 2026, 2025, 2115, 2027], dtype=int64)

In [129]:
tmdb_movies['release_year'] = tmdb_movies['release_date'].map(lambda x: x[:4])
tmdb_movies['release_year'].dtype

dtype('O')

In [130]:
tmdb_movies['release_year'] = tmdb_movies['release_year'].astype('int64')

In [131]:
df = tmdb_movies[(tmdb_movies['release_year']>2010) & (tmdb_movies['release_year']!=2020)]

In [132]:
df['release_year'].unique()

array([2012, 2011, 2014, 2013, 2015, 2017, 2016, 2018, 2019], dtype=int64)

In [133]:
len(df) #23,884

23884

In [134]:
df.head()

genre_ids     id original_language             original_title  \
258             [18, 35]  39356                en                        Boy   
289             [35, 18]  46829                en           Barney's Version   
386                 [18]  54602                en                  Skateland   
409         [28, 18, 53]  56812                es  Balada triste de trompeta   
470  [53, 18, 10749, 80]  47261                es                   Carancho   

     popularity release_date             title  vote_average  vote_count  \
258       7.759   2012-03-02               Boy           7.5         149   
289       7.357   2011-01-14  Barney's Version           7.2         210   
386       5.938   2011-05-13         Skateland           5.4          16   
409       5.552   2011-08-19   The Last Circus           6.7         177   
470       4.554   2011-02-11          Carancho           6.6          59   

     release_year  
258          2012  
289          2011  
386          2011  
409          2011  
470          2011

### Checking for duplicates

In [135]:
df[df.duplicated()]

genre_ids      id original_language  \
2803              [35, 18]   46829                en   
2919                  [18]   54602                en   
2946          [28, 18, 53]   56812                es   
3030   [53, 18, 10749, 80]   47261                es   
3184       [10749, 35, 18]   70588                en   
...                    ...     ...               ...   
26481             [35, 18]  270805                en   
26485             [27, 53]  453259                en   
26504         [27, 35, 27]  534282                en   
26510                 [99]  495045                en   
26511                 [99]  492837                en   

                           original_title  popularity release_date  \
2803                     Barney's Version       7.357   2011-01-14   
2919                            Skateland       5.938   2011-05-13   
2946            Balada triste de trompeta       5.552   2011-08-19   
3030                             Carancho       4.554   2011-02-11   
3184   The Myth of the American Sleepover       3.066   2011-07-22   
...                                   ...         ...          ...   
26481                       Summer League       0.600   2013-03-18   
26485              Devils in the Darkness       0.600   2013-05-15   
26504                                Head       0.600   2015-03-28   
26510                          Fail State       0.600   2018-10-19   
26511                   Making Filmmakers       0.600   2018-04-07   

                                    title  vote_average  vote_count  \
2803                     Barney's Version           7.2         210   
2919                            Skateland           5.4          16   
2946                      The Last Circus           6.7         177   
3030                             Carancho           6.6          59   
3184   The Myth of the American Sleepover           6.2          47   
...                                   ...           ...         ...   
26481                       Summer League           4.0           3   
26485              Devils in the Darkness           3.5           1   
26504                                Head           1.0           1   
26510                          Fail State           0.0           1   
26511                   Making Filmmakers           0.0           1   

       release_year  
2803           2011  
2919           2011  
2946           2011  
3030           2011  
3184           2011  
...             ...  
26481          2013  
26485          2013  
26504          2015  
26510          2018  
26511          2018  

[990 rows x 10 columns]

It looks like there is 990 rows that are duplicated. I will check a couple id numbers to make sure that the duplicated method is running correctly.

In [136]:
df[df['id']==46829]

genre_ids     id original_language    original_title  popularity  \
289   [35, 18]  46829                en  Barney's Version       7.357   
2803  [35, 18]  46829                en  Barney's Version       7.357   

     release_date             title  vote_average  vote_count  release_year  
289    2011-01-14  Barney's Version           7.2         210          2011  
2803   2011-01-14  Barney's Version           7.2         210          2011

In [137]:
df[df['id']==54602]

genre_ids     id original_language original_title  popularity  \
386       [18]  54602                en      Skateland       5.938   
2919      [18]  54602                en      Skateland       5.938   

     release_date      title  vote_average  vote_count  release_year  
386    2011-05-13  Skateland           5.4          16          2011  
2919   2011-05-13  Skateland           5.4          16          2011

### Dropping duplicates

In [138]:
df = df.drop_duplicates(keep='first')

In [139]:
df[df['id']==46829]

genre_ids     id original_language    original_title  popularity  \
289  [35, 18]  46829                en  Barney's Version       7.357   

    release_date             title  vote_average  vote_count  release_year  
289   2011-01-14  Barney's Version           7.2         210          2011

### Checking for missing/placeholder values

In [140]:
df.isna().sum()

genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
release_year         0
dtype: int64

In [141]:
df[df['title']==None]

Empty DataFrame
Columns: [genre_ids, id, original_language, original_title, popularity, release_date, title, vote_average, vote_count, release_year]
Index: []

### EDA to figure out which financial df to use, looking at sample size 

In [142]:
len(tn_movie_budgets) #5782
# len(bom_movie_gross) #3387

5782

In [143]:
release_profit = pd.merge(df, bom_movie_gross, how='inner', left_on='title', right_on='title')
len(release_profit)

2189

In [144]:
release_profit = pd.merge(df, tn_movie_budgets, how='inner', left_on='title', right_on='movie')
len(release_profit)

1903

In [145]:
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [146]:
# tn_movie_budgets[tn_movie_budgets['movie'].map(lambda x: x.startswith('Harry Potter'))]

In [148]:
tn_movie_budgets['worldwide_gross'] = tn_movie_budgets['worldwide_gross'].map(lambda x: x[1:].replace(',',''))

In [153]:
tn_movie_budgets['worldwide_gross'] = tn_movie_budgets['worldwide_gross'].astype('int64')

In [155]:
tn_movie_budgets['worldwide_gross'].dtype

dtype('int64')

In [156]:
def convertdollarstoint(df, col):
    df[col] = df[col].map(lambda x: x[1:].replace(',',''))
    df[col] = df[col].astype('int64')
    return df

In [157]:
convertdollarstoint(df = tn_movie_budgets, col = 'domestic_gross')

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget  domestic_gross  worldwide_gross  
0         $425,000,000       760507625       2776345279  
1         $410,600,000       241063875       1045663875  
2         $350,000,000        42762350        149762350  
3         $330,600,000       459005868       1403013963  
4         $317,000,000       620181382       1316721747  
...                ...             ...              ...  
5777            $7,000               0                0  
5778            $6,000           48482           240495  
5779            $5,000            1338             1338  
5780            $1,400               0                0  
5781            $1,100          181041           181041  

[5782 rows x 6 columns]

In [158]:
convertdollarstoint(df = tn_movie_budgets, col = 'production_budget')

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  
0             425000000       760507625       2776345279  
1             410600000       241063875       1045663875  
2             350000000        42762350        149762350  
3             330600000       459005868       1403013963  
4             317000000       620181382       1316721747  
...                 ...             ...              ...  
5777               7000               0                0  
5778               6000           48482           240495  
5779               5000            1338             1338  
5780               1400               0                0  
5781               1100          181041           181041  

[5782 rows x 6 columns]

In [159]:
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0          425000000       760507625       2776345279  
1          410600000       241063875       1045663875  
2          350000000        42762350        149762350  
3          330600000       459005868       1403013963  
4          317000000       620181382       1316721747

In [161]:
tn_movie_budgets['profit'] = tn_movie_budgets['worldwide_gross'] - tn_movie_budgets['production_budget']
tn_movie_budgets['profit %'] = (((tn_movie_budgets['worldwide_gross'] - tn_movie_budgets['production_budget'])/tn_movie_budgets['production_budget'])*100).round(0)
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross      profit  profit %  
0          425000000       760507625       2776345279  2351345279     553.0  
1          410600000       241063875       1045663875   635063875     155.0  
2          350000000        42762350        149762350  -200237650     -57.0  
3          330600000       459005868       1403013963  1072413963     324.0  
4          317000000       620181382       1316721747   999721747     315.0

In [164]:
tn_movie_budgets['release_month'] = tn_movie_budgets['release_date'].map(lambda x: x[:3])

In [165]:
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross      profit  profit %  \
0          425000000       760507625       2776345279  2351345279     553.0   
1          410600000       241063875       1045663875   635063875     155.0   
2          350000000        42762350        149762350  -200237650     -57.0   
3          330600000       459005868       1403013963  1072413963     324.0   
4          317000000       620181382       1316721747   999721747     315.0   

  release_month  
0           Dec  
1           May  
2           Jun  
3           May  
4           Dec